src/react-forms.js

```diff
@@ -416,20 +416,24 @@ class ReactForms extends Component {
     }
 
     this.blockSubmission = true;
-    await this.startSubmit();
+    try {
+      await this.startSubmit();
 
-    const maybePromisedErrors = this.runValidations();
+      const maybePromisedErrors = this.runValidations();
 
-    if (isPromise(maybePromisedErrors)) {
-      const errors = await maybePromisedErrors;
-      await this.setErrors(errors, false, true);
-    } else {
-      await this.setErrors(maybePromisedErrors, false, true);
-    }
+      if (isPromise(maybePromisedErrors)) {
+        const errors = await maybePromisedErrors;
+        await this.setErrors(errors, false, true);
+      } else {
+        await this.setErrors(maybePromisedErrors, false, true);
+      }
 
-    const result = await this.executeSubmit();
-    this.blockSubmission = false;
-    return result;
+      const result = await this.executeSubmit();
+      this.blockSubmission = false;
```

Comment:
Could shove this in a finally block 🤷‍♀

Reply:
Actually good point is it better to shove this in a catch-less finally block and let the error fall through? Are we just ignoring submit errors now? @julianburr

Reply:
you can't bail on catch/finally if you use async/await, or am I misunderstanding what you guys mean?

Reply:
or do you mean use `.finally` where we execute `this.submitForm`? In general finally is not (yet) part of the promise spec, so unless you want to polyfill which I wouldn't do on lib level I wouldn't ue it 🤔

Reply:
> Are we just ignoring submit errors now

what do you mean?

Reply:
Are we not just swallowing submit errors now with that `catch` block? I would imagine we still want to handle them in in our apps, or I'm just not getting how the errors propagate in this case...

My thoughts are 
```
try {
  ...
  const result = await this.executeSubmit();
  ...
} finally {
  this.blockSubmission = false
}
```

This way errors in the `handleSubmit` function won't be swallowed and we still unblock submission.

Reply:
sounds good to me, can you change it? :blush:

Reply:
Yep will do